commit 93943c9ec94879c4373e19341ccc1d05dc16afd2
Author: Peter Collingbourne <pcc@google.com>
Date:   Wed May 15 20:28:08 2013 +0200

    Move _obstack_compat out of common
    
    it is impossible to create an alias of a common symbol (as
    compat_symbol does), because common symbols do not have a section or
    an offset until linked.  GNU as tolerates aliases of common symbols by
    simply creating another common symbol, but other assemblers (notably
    LLVM's integrated assembler) are less tolerant.
    
    2013-05-15  Peter Collingbourne  <pcc@google.com>
    
            * malloc/obstack.c (_obstack_compat): Add initializer.
    -

diff --git a/malloc/obstack.c b/malloc/obstack.c
index 5786da0aa4..c27a422077 100644
--- a/malloc/obstack.c
+++ b/malloc/obstack.c
@@ -116,7 +116,7 @@ int obstack_exit_failure = EXIT_FAILURE;
 /* A looong time ago (before 1994, anyway; we're not sure) this global variable
    was used by non-GNU-C macros to avoid multiple evaluation.  The GNU C
    library still exports it because somebody might use it.  */
-struct obstack *_obstack_compat;
+struct obstack *_obstack_compat = 0;
 compat_symbol (libc, _obstack_compat, _obstack, GLIBC_2_0);
 #  endif
 # endif
